File: ruby/high-scores/high_scores.rb
class HighScores

    def initialize(scores)
        @scores = scores
    end

    def scores
        return @scores
    end

    def latest
        return @scores[-1]
    end

    def personal_best
        return @scores.max
    end

    def personal_top_three
        sorted = @scores.sort.reverse
        top_three = []
        top_three.push(sorted[0])
        top_three.push(sorted[1]) if sorted[1] != nil
        top_three.push(sorted[2]) if sorted[2] != nil
        return top_three
    end

end

File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    List<int> _highScores;
    public HighScores(List<int> list)
    {
        _highScores = list;
    }

    public List<int> Scores()
    {
        return _highScores;
    }

    public int Latest()
    {
        return _highScores[_highScores.Count - 1];
    }

    public int PersonalBest()
    {
        return _highScores.Max();
    }

    public List<int> PersonalTopThree()
    {
        // List<int> personalTopThreeScores = new List<int> {};
        // personalTopThreeScores.Add(_highScores.Max());
        // _highScores.Remove(_highScores.Max());
        // if (_highScores.Count > 0)
        // {
        //     personalTopThreeScores.Add(_highScores.Max());
        //     _highScores.Remove(_highScores.Max());
        // }
        // if (_highScores.Count > 0)
        // {
        //     personalTopThreeScores.Add(_highScores.Max());
        //     _highScores.Remove(_highScores.Max());
        // }
        // return personalTopThreeScores;

        return _highScores.OrderByDescending(x => x).Take(3).ToList();
    }
}